spk/hassio/src/service-setup.sh

```diff
@@ -0,0 +1,55 @@
+CFG_FILE="${SYNOPKG_PKGDEST}/etc/hassio.json"
+
+service_preinst() {
+    exit 0
+}
+
+service_postinst() {
+    DATA_DIR="${share_path}/${folder_name}"
+
+    URL_VERSION="https://s3.amazonaws.com/hassio-version/stable.json"
+
+    HOMEASSISTANT_DOCKER="homeassistant/qemux86-64-homeassistant"
+    HASSIO_DOCKER="homeassistant/amd64-hassio-supervisor"
+
+    # Read infos from web
+    HASSIO_VERSION=$(curl -s $URL_VERSION | jq -e -r '.supervisor')
+
+    # Pull supervisor image
+    /usr/local/bin/docker pull "$HASSIO_DOCKER:$HASSIO_VERSION" >/dev/null
```

Comment:
This process of pulling, tagging the supervisor image, would it be difficult to move this logic to a separate file, and run it both here and when the service restarts?  This would allow it to handle things like a timeout blocking the pull, installation finishes, but the image is unusable -- by re-checking on restart, it would allow a user to retry by restarting.  It might also get around the issue of upgrading a `hassio` when the `homeassistant` isn't running causing no `homeassistant` to be running after re-installing.

Reply:
I guess we could. But that is not the way hassio normally works.

Reply:
How should the test for this look? If i move it to `hassio.sh` I mean, hassio is responsible to update itself. But if this fails the first time we will have an issue. Wouldn't it be easier to just fail the installation?

Reply:
Hey @fredrike , what I was saying is "there is a tagging logic that pulls, tags, runs the supervisor image.  maybe also do it when the service restarts?" ; additionally: "this can moved to a separate script and called in two places to avoid maintaining copies of the same logic".  That is to say, if a script present in the SPK does the work of pulling, tagging an image, and is run from two places (current and in a restart) then easier to maintain.  I wasn't suggesting to have the script added to hassio itself and downloaded.

This could also be done as a shell-function, or as duplicate lines of code in two places.  Whichever.  Either way, because the reused function/behavior is present int hr SPK, a failed download would be remediable by restarting the SPK, which would effectively re-pull, re-tag, and start the supervisor.